TuviSRPLib/TuviSRPLib/ExtendedHashDigest.cs

```diff
@@ -46,18 +48,17 @@ public void BlockUpdate(byte[] input, int inOff, int inLen)
                 throw new ArgumentOutOfRangeException(nameof(inLen), "Parameter inLen can not be negative.");
             }
 
-            byte[] newMessage = new byte[_message.Length + inLen];
-            Array.Copy(_message, newMessage, _message.Length);
-            Array.Copy(input, inOff, newMessage, _message.Length, inLen);
-            _message = newMessage;
+            Array.Resize(ref _message, _message.Length + input.Length);
+            Array.Copy(input.ToArray(), 0, _message, _message.Length - input.Length, input.Length);
         }
 
 #if NETCOREAPP2_1_OR_GREATER || NETSTANDARD2_1_OR_GREATER
         /// <summary>Update the message digest with a span of bytes.</summary>
         /// <param name="input">the span containing the data.</param>
         public void BlockUpdate(ReadOnlySpan<byte> input)
         {
-             throw new NotImplementedException();
+            Array.Resize(ref _message, _message.Length + input.Length);
```

Comment:
this code doesn't do less allocations,
look at https://github.com/dotnet/runtime/blob/8ce4279520358f344ddd5c82d8f66cc653cf9a4f/src/libraries/System.Private.CoreLib/src/System/Array.cs#L43

We need our own method here which will do an allocation ONLY if we don't have enough memory in `_message` AND in case of reallocation we have to Zero previous buffer